src/app/services/task.service.spec.ts

```diff
@@ -0,0 +1,149 @@
+// @angular
+import { TestBed } from '@angular/core/testing';
+
+// services
+import { TaskService } from './task.service';
+import { WebService } from './web.service';
+
+// models
+import { Task } from '../models/task.model';
+
+// mocks
+import { WebServiceMock } from '../mocks/mock.web.service';
+
+let taskService: TaskService;
+let webService: WebService;
+
+const task: Task = { name: 'task 1', isCompleted: true};
+const tasks: Task[] = [
+  { name: 'task 1', isCompleted: true },
+  { name: 'task 2', isCompleted: false },
+  { name: 'task 3', isCompleted: false },
+  { name: 'task 4', isCompleted: true },
+];
+
+describe('TaskService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        TaskService,
+        { provide: WebService, useClass: WebServiceMock }
+      ]
+    });
+    taskService = TestBed.get(TaskService);
+    webService = TestBed.get(WebService);
+    taskService.tasks = tasks;
+  });
+
+  it('should be created', () => {
+    expect(taskService).toBeTruthy();
+  });
+
+  it('call renderAllTasks() => do not call WebService.getTasks() if local storage return array of tasks', () => {
```

Comment:
1) `it` description text is extension of `describe` text, it will be one phrase when test running. For example `TaskService` `should be created` or `TaskService` `should get data from storage`
2) Hard to understand descriptions, it should have contain only text that describes functionality

Reply:
done using L. Nesvitiy suggestions )